### ISSUE TITLE
gdscript LSP active on other filetypes

### ISSUE THREAD

**Issue by ray-pH** [grndctrl/zed-gdscript#6]
When GDscript is running, the LSP is also active on other filetypes

Is there a config I should add to limit it to only run on GDScript files?

![image](https://github.com/grndctrl/zed-gdscript/assets/57003984/688964ad-b07b-478e-9ba6-5c5039be51de)
![image](https://github.com/grndctrl/zed-gdscript/assets/57003984/c07cfdbd-7648-4ac2-b1c5-f745726d2a27)

**Comment by grndctrl** [comment]
That is weird, GDScript shouldn't even be triggered in those files.

**Comment by ray-pH** [comment]
It only happens after the gdscipt lsp is running. Without opening a gdscript file first, the lsp won't be active.

**Comment by grndctrl** [comment]
I'm think this is more of a Zed bug. It shouldn't check `.rs` files against the GDScript LSP. The extension is configured to only work on GDScript files (and resource files, since last update)

**Comment by grndctrl** [comment]
could you run `copy system specs into clipboard` in the command palette inside Zed, and paste it here? I will open an issue on Zed repo.

**Comment by ray-pH** [comment]
Zed: v0.141.0 (Zed Dev 4855da53dfa62791d9dd102b9db9cf7ddb08d153)
OS: Linux Wayland debian unknown
Memory: 13.5 GiB
Architecture: x86_64

Might be an issue with the linux port?

**Comment by coppolaemilio** [comment]
I have a similar issue, where the Project Diagnostics is flagging gdscript issues on the `.tres` files:
![image](https://github.com/grndctrl/zed-gdscript/assets/2206700/36ff77be-aba0-4e6c-880c-ff0b364f03cd)
I'm on MacOs Sonoma 14.5, M1 Pro, latest Zed version (Zed 0.141.2 3e609e479354cfb623e001ea24a559351607076f)

**Comment by EnderMaster77** [comment]
Experiencing the same issue.
Zed: v0.146.3 (Zed)
OS: Linux Wayland arch unknown
Memory: 31.3 GiB
Architecture: x86_64
GPU: NVIDIA GeForce GTX 1650 SUPER || NVIDIA || 555.58.02

**Comment by NathanLovato** [comment]
@coppolaemilio Do you still have this issue with the latest release of Zed? On this end, I cannot reproduce it.


I can reproduce the issue of the language server running on other programming language files, though. It happens with any programming language it seems. I'm not sure why yet, but I'll try to see if I can find out. If anyone has an idea of what could be the cause, I'm all ears. I'll look into language injections and language server settings, see if something stops the behavior.

**Comment by coppolaemilio** [comment]
Nope, can't reproduce it anymore. It seems fixed on my end 👍

**Comment by NathanLovato** [comment]
@coppolaemilio Yes! And the idea is to develop it a bit and then if more people start to use Zed offer to transfer it to the foundation like the emacs plugin before.

**Comment by grndctrl** [comment]
> Nope, can't reproduce it anymore. It seems fixed on my end 👍 btw, did you take over the project? @NathanLovato 👀

Yeah, I threw together the extension just so I could keep working in Zed. But then people started using it and I was unable to keep up. Glad @NathanLovato reached out!

**Comment by fstxz** [comment]
I am still having this issue (including .tres files). Linux, Zed 0.148.0

**Comment by eth0net** [comment]
I got around this issue myself by adding the following setting in Zed ^^
```
  "languages": {
    "Markdown": {
      "language_servers": ["!gdscript", "..."]
    }
  },
```
It does however only solve the issue one language at a time 👀

**Comment by eth0net** [comment]
Just realised this is also possible... this works xD
```
  "language_servers": ["!gdscript", "..."],
  "languages": {
    "GDScript": {
      "language_servers": ["gdscript", "..."]
    },
  },
```

**Comment by KettLovahr** [comment]
Hello, I have this issue as well, here's the language server reporting a bunch of issues on my tasks.json file:

![An image showing several gdscript related errors on a json file](https://github.com/user-attachments/assets/217634d8-541b-461c-9431-98cc67b450c7)

This happens when:

1. The language server is active (Godot is running)
2. You save any text file in the workspace (that's not a .gd file)

Zed: v0.164.2 (Zed)
OS: Linux Wayland opensuse-tumbleweed 20250112
Memory: 15.5 GiB
Architecture: x86_64
GPU: AMD Radeon RX 6600 (RADV NAVI23) || radv || Mesa 24.3.3

**Comment by Na-r** [comment]
I submitted a pull request fixing this into Godot Engine. Thanks @KettLovahr for the detailed response!

It seems like Zed used to not send the notification for all file types, but it was reverted in zed-industries/zed/pull/19183, so I submitted the fix into the engine.

**Comment by tkgalk** [comment]
<img width="1190" height="581" alt="Image" src="https://github.com/user-attachments/assets/d84bcd49-369f-4b93-99f6-a658a1415bc6" />

Still a problem it seems.
